miner/worker.go

```diff
@@ -935,6 +939,18 @@ func (w *worker) prepareWork(genParams *generateParams) (*environment, error) {
 		log.Error("Failed to create sealing context", "err", err)
 		return nil, err
 	}
+	if w.chainConfig.IsCancun(header.Number, header.Time) {
+		if header.BeaconRoot == nil {
```

Comment:
I think we need to decide if this is the right place for this check. We *could* check it earlier, i.e. just require that the `beaconRoot` be submitted. However, checking it here has the advantage that the consensus engine can override it. This will mostly be useful for downstream projects, who might want to set this there.

Reply:
So the options are ... 

1. Leave it as is, 
2. Remove the check (potentially getting a nil deref on `core.ProcessBeaconBlockRoot(*header.BeaconRoot,`
3. Change the check to use `common.Hash{}` if it is nil, 
4. Change the check to not `ProcessBeaconBlockRoot` at all, if it is nil

But perhaps you were thinking of something else? 

I mean, whatever we somewhere else, if we deref a pointer here, we should check it first, so IMO the only question is if we barge ahead or back out. 

Out of the options I listed, I'd prefer `1`, possibly `4` for flexibility, but that's the kind of flexibility which can bite us later on.

Reply:
I decided to go with (4)